### PR TITLE
Printcolumn added in VSphereCluster and VSphereMachine

### LINKED ISSUE #572: Print fields as columns with VSphereCluster and VSphereMachine resources
/kind feature

**Describe the solution you'd like**
CAPI recently started annotating certain fields in the `Cluster` and `Machine` resources so that the marked fields will be printed as columns when executing `kubectl get clusters` or `kubectl get machines`. For example:

```shell
$ kubectl get clusters
NAME                 PHASE
management-cluster   provisioned
workload-cluster-1   provisioned
```

```shell
$ kubectl get machines
NAME                                       PROVIDERID                                       PHASE
management-cluster-controlplane-0          vsphere://4230e346-b34f-708f-2d7c-e680d433f868   running
workload-cluster-1-controlplane-0          vsphere://423093f7-6e8f-9fa0-f092-ebb3d3097b92   running
workload-cluster-1-md-0-689c674fc4-q26b5   vsphere://42304a73-f633-57d3-f9ae-1d020473681b   running
```

This is accomplished by adding `kubebuilder` annotations as Godocs to the resources' analogous Go types. For example:

**The CAPI [`Cluster` annotation](https://github.com/kubernetes-sigs/cluster-api/blob/5cd7261c53109d1af832c2f26eb969cfe4b8e9a0/api/v1alpha2/cluster_types.go#L151-L154)**

```golang
// +kubebuilder:printcolumn:name="Phase",type="string",JSONPath=".status.phase",description="Cluster status such as Pending/Provisioning/Provisioned/Deleting/Failed"

// Cluster is the Schema for the clusters API
type Cluster struct {
```

**The CAPI [`Machine` annotation](https://github.com/kubernetes-sigs/cluster-api/blob/5cd7261c53109d1af832c2f26eb969cfe4b8e9a0/api/v1alpha2/machine_types.go#L199-L204)**

```golang
// +kubebuilder:printcolumn:name="ProviderID",type="string",JSONPath=".spec.providerID",description="Provider ID"
// +kubebuilder:printcolumn:name="Phase",type="string",JSONPath=".status.phase",description="Machine status such as Terminating/Pending/Running/Failed etc"
// +kubebuilder:printcolumn:name="NodeName",type="string",JSONPath=".status.nodeRef.name",description="Node name associated with this machine",priority=1

// Machine is the Schema for the machines API
type Machine struct {
```

This issue discusses whether or not CAPV should add annotations to its `VSphereCluster` and `VSphereMachine` resources to print specific fields as columns when users execute `kubectl get vsphereclusters` and `kubectl get vspheremachines`. The questions are:

* Should we do it?
  * For `VSphereCluster`?
    * Which fields?
  * For `VSphereMachine`?
    * Which fields?

cc @andrewsykim @sidharthsurana

### PR DESCRIPTION
**Pull request by evalsocket**
**What this PR does / why we need it**:
Added print column for  VSphereCluster and VSphereMachine

**Which issue(s) this PR fixes** :
Fixes #572 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by evalsocket** [comment]
@randomvariable PR is ready for review

File: api/v1alpha3/vspherecluster_types.go [hunk 1]
```diff
@@ -70,6 +70,10 @@ type VSphereClusterStatus struct {
 // +kubebuilder:resource:path=vsphereclusters,scope=Namespaced,categories=cluster-api
 // +kubebuilder:storageversion
 // +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels.cluster\\.x-k8s\\.io/cluster-name",description="Cluster to which this VSphereCluster belongs"
+// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.ready",description="Cluster infrastructure is ready for VSphereMachine"
+// +kubebuilder:printcolumn:name="Server",type="string",JSONPath=".spec.server",description="Server is the address of the vSphere endpoint."
+// +kubebuilder:printcolumn:name="ControlPlaneEndpoint",type="string",JSONPath=".status.controlPlaneEndpoint[0]",description="API Endpoint",priority=1
```
Comment (yastij) [inline api/v1alpha3/vspherecluster_types.go:76]: the `controlPlaneEndpoint` isn't the under status it's on the spec

File: api/v1alpha3/vspherecluster_types.go [hunk 2]
```diff
@@ -70,6 +70,10 @@ type VSphereClusterStatus struct {
 // +kubebuilder:resource:path=vsphereclusters,scope=Namespaced,categories=cluster-api
 // +kubebuilder:storageversion
 // +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels.cluster\\.x-k8s\\.io/cluster-name",description="Cluster to which this VSphereCluster belongs"
```
Comment (yastij) [inline api/v1alpha3/vspherecluster_types.go:73]: not sure this is useful, most times the vspherecluster and cluster name are matching

File: api/v1alpha3/vspheremachine_types.go [hunk 3]
```diff
@@ -100,6 +100,10 @@ type VSphereMachineStatus struct {
 // +kubebuilder:resource:path=vspheremachines,scope=Namespaced,categories=cluster-api
 // +kubebuilder:storageversion
 // +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels.cluster\\.x-k8s\\.io/cluster-name",description="Cluster to which this VSphereMachine belongs"
+// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.ready",description="Machine ready status"
+// +kubebuilder:printcolumn:name="ProviderID",type="string",JSONPath=".spec.providerID",description="VSphereMachine instance ID"
+// +kubebuilder:printcolumn:name="Machine",type="string",JSONPath=".metadata.ownerReferences[?(@.kind==\"Machine\")].name",description="Machine object which owns with this VSphereMachine"
```
Comment (yastij) [inline api/v1alpha3/vspheremachine_types.go:106]: I would add a priority 1 here

**Review by yastij — COMMENTED** [review]
can you share an output of how this looks like once the comments are addressed ?

**Comment by evalsocket** [comment]
@yastij sorry for the delay. I deployed the CRD but not created a cluster. I don't have vsphere setup. Do you know any way to print header without creating cluster

**Comment by gab-satchi** [comment]
Fixes #1119

**Comment by srm09** [comment]
@evalsocket Please squash the commits and fill the **release note** section appropriately.

**Comment by srm09** [comment]
@evalsocket Would you be able to squash the commits, and then the PR is good to go.

**Comment by srm09** [comment]
@evalsocket Do you mind changing the base branch for the PR to `release-0.7` instead, which will make this change a part of the v1alpha3 API types? Also could you squash the commits?

**Pull request closed without merging**